Configure heartbeat in the kafka connect connectors

welansari opened the pull request:
Hearbeat prevents having an outdated resume token in the connectors by constantly updating the offset to the last object in the oplog

Outdated offsets can happen in two cases:
- Not enough updates in MongoDB, so the offset stays the same until the oplog event holding the offset gets expired from the oplog.
- Too many updates in the DB, which causes the event holding the offset to be overwritten as the oplog is a capped collection that overwrites the earliest events when it fills up.

Issue: BB-407

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-407 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.6.21`


Please check the `Fix Version/s` of BB-407, or the target
branch of this pull request.

francoisferrand commented on extensions/oplogPopulator/modules/ConnectorsManager.js:
```diff
@@ -82,7 +82,11 @@ class ConnectorsManager {
+            }),
+            // hearbeat prevents having an outdated resume token in the connectors
+            // by constantly updating the offset to the last object in the oplog
+            'heartbeat.interval.ms': process.env.OPLOG_HEARTBEAT_INTERVAL_MS ?
```
should this not be configured via backbeat config file instead?

Commit:
Configure heartbeat in the kafka connect connectors

Hearbeat prevents having an outdated resume token in the connectors
by constantly updating the offset to the last object in the oplog

Issue: BB-407

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand reviewed (approved):
LGTM

should we add the param to the `https://github.com/scality/backbeat/blob/HEAD/conf/config.json`, and possibly add tests?

welansari commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.6`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.4`
* `development/8.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.6`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.4`
* `development/8.5`



Please check the status of the associated issue BB-407.

Goodbye kerkesni.
